mod-source-record-storage-server/src/main/java/org/folio/services/RecordServiceImpl.java

```diff
@@ -170,6 +175,22 @@ public Future<ParsedRecordsBatchResponse> updateParsedRecords(RecordCollection r
     return recordDao.updateParsedRecords(recordCollection, tenantId);
   }
 
+  @Override
+  public Future<RecordCollection> fetchParsedRecords(FetchParsedRecordsBatchRequest fetchRequest, String tenantId) {
+    var ids = fetchRequest.getConditions().getIds();
+    var idType = IdType.valueOf(fetchRequest.getConditions().getIdType());
+    if (ids.isEmpty()) {
+      Promise<RecordCollection> promise = Promise.promise();
+      promise.complete(new RecordCollection().withTotalRecords(0));
+      return promise.future();
+    }
+
+    var idsCondition = getExternalIdsCondition(ids, idType);
+    var recordType = toRecordType(fetchRequest.getRecordType().name());
+    return recordDao.getRecords(idsCondition, recordType, emptyList(), 0, ids.size(), tenantId)
```

Comment:
I believe that this method in `recordDao` do to many operations, and we need the endpoint to be as performance as possible.
There is a lot of stuff that clould be excluded from the response. Like: metadata, additionalInfo, errorRecord, rawRecord and so on. We need just parsed content and ids.

Reply:
Done. Added a new Dto